h2o-core/src/main/java/water/AutoBuffer.java

```diff
@@ -139,7 +144,17 @@
     _read = true;               // Reading by default
     _firstPage = true;
     // Read Inet from socket, port from the stream, figure out H2ONode
-    _h2o = H2ONode.intern(remoteAddress, getPort());
+    if(remoteAddress!=null) {
+      _h2o = H2ONode.intern(remoteAddress, getPort());
+    }else{
+      // In case the communication originates from non-h2o node, we set _h2o node to null.
+      // It is done for 2 reasons:
+      //  - H2ONode.intern creates a new thread and if there's a lot of connections
+      //    from non-h2o environment, it could end up with too many open files exception.
+      //  - H2OIntern also reads port (getPort()) and additional information which we do not send
+      //    in communication originating from non-h2o nodes
+      _h2o = null;
```

Comment:
There are asserts in the code that expect _h2o field to be set. I feel like it might be a bad idea to use AutoBuffer to communicate with non-H2O.